### ISSUE TITLE
Compilation errors - failed to run custom build command

### ISSUE THREAD

**Issue by Ploppz** [sfackler/rust-openssl#992]
I just started an empty crate with this as the only dependency: https://github.com/openai/gym-http-api/tree/master/binding-rust

Compilation fails when compiling rust-openssl apparently: https://bpaste.net/show/768f4f030265

**Comment by sfackler** [comment]
You're using a very old version of this crate which doesn't support OpenSSL 1.1.0. You should upgrade to a newer hyper version.

**Comment by crumblingstatue** [comment]
OpenSSL 1.1.0  breaks a bunch of things that are dependent on rust-openssl, including `native-tls -> reqest -> cargo-edit`.

Should I just open issues/pull-requests to upgrade their rust-openssl dependencies?

Does the new rust-openssl crate still support old OpenSSL versions along with 1.1.0?

**Comment by sfackler** [comment]
All versions of native-tls support OpenSSL 1.1.0.

**Comment by crumblingstatue** [comment]
I apologize, I assumed this issue is talking about the error that I am experiencing, but it's actually a different error.

The error I'm experiencing seems to be #986.

**Comment by Ploppz** [comment]
Well that particular error was gone when I changed to latest `hyper`. I did get some errors from the `bindings-rust` crate itself since it was written for the old `hyper`. I don't know whether those errors would appear _after_ the errors I presented in the original post. If that is the case then the issue resolved by using latest `hyper`. Anyhow, I decided to not use `bindings-rust` after all (because it's so outdated).
